base/set.jl

```diff
@@ -625,10 +625,14 @@ end
 function _replace!(new::Callable, res::AbstractArray, A::AbstractArray, count::Int)
     c = 0
     if count >= length(A) # simpler loop allows for SIMD
```

Comment:
Making the same change when `count < length(A)` doesn't seem to lead to benefits, on the contrary (at least on my machine).